source/libs/sync/src/syncMain.c

```diff
@@ -2298,6 +2299,12 @@ int32_t syncNodeOnHeartbeat(SSyncNode* ths, const SRpcMsg* pRpcMsg) {
   pMsgReply->startTime = ths->startTime;
   pMsgReply->timeStamp = tsMs;
 
+  if(CID(&(pMsg->srcId)) != CID(&(ths->myRaftId)))
+  {
+    sWarn("vgId:%d, drop heartbeat msg from dnode:%d, because it come from another cluster:%d, differ from current cluster:%d", ths->vgId, DID(&(pMsg->srcId)), CID(&(pMsg->srcId)) , CID(&(ths->myRaftId)));
+    return 0;
```

Comment:
There is a potential memory leak here, since rpcMsg.pCont is not released or transferred before returning.